### DISCUSSION TITLE
How to return the dynamic route's statically generated page (getStaticPaths with getStaticProps) for some route values, and use the same static page for everything else

### DISCUSSION THREAD

**Discussion by jstansbe** [vercel/next.js#26635] [Help]
## **What I'm trying to accomplish:**
All statically generated pages from (getStaticPaths + getStaticProps) return the respective static file, any other request gets served from a single static page.

#### **Example:**
Lets say I have a twitter clone called, "mypage" that has a url of mypage.com with a dynamic route/page:  `mypage.com/[username]` (the pages folder contains a file named: `[username].tsx`)

I want the vercel and next.js accounts to have their pages pre-rendered with data from their respective rows in my db (ie username, handle, bio, followers count, etc). So if you visit mypage.com/vercel you get served a static page (already rendered with data for vercel account from db). If  you mypage.com/nextjs you get served a static page (already rendered with data for nextjs account from db). This could be the case for thousands of other accounts.... then you visit mypage.com/jstansbe and you get served a generic static page (generic-user-page component) that has no pre-rendered content specific to jstansbe. The page is loaded in the users browser, then a client-side api request is sent to get jstansbe's data from the db, the data is then received and the page renders all elements/components.

## **What I've tried**
I've been trying to find a way to use the fallback prop, but all it does is either serve a 404 page or create a situation where a static page specific to the passed route val (in example above: username) is created and then saved to the server. I've looked into getServerSideProps which I can serve up a generic page to handle all users (usernames) routes but can't serve a subset of users with statically generated pages populated with their respective database data. 

This is the core issue, I only want statically generated files for users (username) included in the getStaticPaths to be saved to the server. All other requests would use the generic component (generic-user-page component) that makes the request for the user data client-side (aka from the devices browser).

Any ideas or thoughts on this would be appreciated. If this isn't possible I'm more than happy to convert/reopen as a feature request.

**Comment by atawfique21** [comment]
I'm also trying to accomplish this and would love input from anyone who has it. 

I'm creating a similar project and if the profile exists in StaticPaths, show that static content. Otherwise, I want to show a 'User Not Found' page. How can I achieve this?

I have tried fallback: 'blocking, true and false. None of them let me use a ternary operator to show an error page if user data does not exist.

**Reply by atawfique21** [reply]
I've found a solution to this in my case and I'm sure you can adapt it to yours @jstansbe. 

You will want to dictate `fallback: true` inside of `getStaticPaths()`

In my situation, I have a **userData** variable inside of `getStaticProps()` which gets data in a function outside of the file. 
For profiles that do not exist, I am doing this: 

```javascript
// Inside getStaticProps() function
  const userData = await getUserData(params.id)

  if (!userData) {
    // return { notFound: true } <-- You can use this to return a generic 404 enabled by Next.js as part of the v10 update.
    return {
      props: {
        userData: 'none' // <-- use this method if you want a custom page, such as OP's request
      }
    }
  }
```

And inside of your page return function, you can use this code to generate that custom page of your choice:

`
if (userData === 'none') {
    return <p>This profile could not be found.</p>
}
`
